### DISCUSSION TITLE
Duplicate Event Logs for app_store_subscription_renew on iOS

### DISCUSSION THREAD

**Discussion by Prakunj** [hyochan/react-native-iap#2820] [Q&A]
We have observed that the app_store_subscription_renew event(firebase automatically collected event) is being triggered during subscription purchase multiple times on iOS. This issue is causing inflated event counts and making it difficult to accurately analyze user subscription behaviour. Please check ss for reference.

![WhatsApp Image 2024-08-19 at 9 00 38 PM](https://github.com/user-attachments/assets/a04e052d-b719-4e40-ab8d-ea0f652091ec)

**Comment by hyochan** [comment]
The `app_store_subscription_renew` event is a **Firebase automatically collected event**, not something triggered by react-native-iap.

## Likely Causes

### 1. Xcode Version Issue

Older Xcode versions (before 16.4) have known issues causing duplicate events.

**Solution**: Upgrade to **Xcode 16.4 or later**.

### 2. Firebase SDK Behavior

Firebase Analytics automatically tracks App Store events. Duplicate logging can occur due to:

- Multiple Firebase initialization
- App lifecycle events triggering multiple callbacks
- Firebase SDK internal behavior

### 3. Sandbox Environment

The iOS Sandbox environment is known to be unreliable and may trigger duplicate events that don't occur in production.

## Solutions

1. **Upgrade Xcode** to 16.4+
2. **Test in Production/TestFlight** - Sandbox can have duplicate event issues
3. **Deduplicate on your server** - Use `transactionId` to identify unique events
4. **Check Firebase setup** - Ensure Firebase is initialized only once

## Deduplication Example

```typescript
const processedTransactions = new Set<string>();

const handlePurchase = (purchase: Purchase) => {
  if (processedTransactions.has(purchase.transactionId)) {
    return; // Already processed
  }
  processedTransactions.add(purchase.transactionId);

  // Process purchase...
};
```

## References

- [Troubleshooting - Xcode Version](https://hyochan.github.io/react-native-iap/docs/guides/troubleshooting)
- [Firebase Auto-Collected Events](https://support.google.com/firebase/answer/9234069)
